test/python/unittest/API/APITest.py

```diff
@@ -0,0 +1,339 @@
+
+        # Get series.
+        series = self.__series
+        self.assertIsInstance(series, openPMD.Series)
+
+        print("Read a Series with openPMD standard version %s" %
+              series.openPMD)
+
+        # Loop over iterations.
+        print("The Series contains {0} iterations:".format(
+            len(series.iterations)))
+        for i in series.iterations:
+            print("\t {0}".format(i))
+        print("")
+
+        # Check type.
+        i = series.iterations[100]
+        self.assertIsInstance(i, openPMD.Iteration)
+
+        # Loop over meshes and particles.
+        print("Iteration 100 contains {0} meshes:".format(len(i.meshes)))
+        for m in i.meshes:
+            print("\t {0}".format(m))
+        print("")
+        # Get a mesh.
+        E = i.meshes["E"]
+        self.assertIsInstance(E, openPMD.Mesh)
+
+        print("Iteration 100 contains {0} particle species:".format(
+            len(i.particles)))
+        for ps in i.particles:
+            print("\t {0}".format(ps))
+        print("")
+
+        # Get a mesh.
+        electrons = i.particles["electrons"]
+        self.assertIsInstance(electrons, openPMD.ParticleSpecies)
+
+    def testLoadSeries(self):
+        """ Test loading a pmd series from hdf5."""
+
+        # Get series.
+        series = self.__series
+        self.assertIsInstance(series, openPMD.Series)
+
+        print("Read a Series with openPMD standard version %s" %
+              series.openPMD)
+
+    def testIterations(self):
+        """ Test querying a series' iterations and loop over them. """
+
+        # Get series.
+        series = self.__series
+
+        # Loop over iterations.
+        self.assertIsInstance(series.iterations, openPMD.Iteration_Container)
+        self.assertTrue(hasattr(series.iterations, "__getitem__"))
+        for i in series.iterations:
+            self.assertIsInstance(i, int)
+            self.assertIsInstance(series.iterations[i], openPMD.Iteration)
+
+        # Check type.
+        i = series.iterations[100]
+        self.assertIsInstance(i, openPMD.Iteration)
+
+    def testMeshes(self):
+        """ Test querying a mesh. """
+
+        # Get series.
+        series = self.__series
+
+        # Check type.
+        i = series.iterations[100]
+
+        # Check meshes are iterable.
+        self.assertTrue(hasattr(i.meshes, "__getitem__"))
+        for m in i.meshes:
+            self.assertIsInstance(m, str)
+            self.assertIsInstance(i.meshes[m], openPMD.Mesh)
+
+    def testParticles(self):
+        """ Test querying a particle species. """
+
+        # Get series.
+        series = self.__series
+
+        # Check type.
+        i = series.iterations[100]
+
+        self.assertTrue(hasattr(i.particles, "__getitem__"))
+        for ps in i.particles:
+            self.assertIsInstance(ps, str)
+            self.assertIsInstance(i.particles[ps], openPMD.ParticleSpecies)
+
+    def testAllocation(self):
```

Comment:
This can be removed together with the C++ [code](https://github.com/openPMD/openPMD-api/blob/06ae64508f726d18eaedc60d7e73580e7f24a054/include/openPMD/RecordComponent.hpp#L52-L57) and [bindings](https://github.com/openPMD/openPMD-api/blob/b95e33c6627aa41afca80bad0d2265846451f5e4/src/binding/python/RecordComponent.cpp#L178-L182) behind it. We do not allow the user to specify allocation strategy any more.
If you include this, I'll remove it when we get rid of the ```enum```.

Reply:
I'll leave it in but modify: now asserts AttributeError on openPMD.Allocation .